### PR TITLE
update: sync outputs with transactions

### LINKED ISSUE #3941: [Task]: Sync outputs with transactions
### Task description

We should attempt to sync the outputs with their containing transaction. This will allow us to show more details on the activity feed.

### Requirements

- Add sync options to restore, background sync and sync to enable transaction syncing
- Build activity from transaction if available instead of output on new output event

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary
Use the transaction inputs (if available) to determine the sender of the output

## Changelog

```
- use additional inputs on NewOutputEvent
- use additional inputs on loadAccounts
```

## Relevant Issues
Closes #3941 

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/profile-manager/api/events/newOutput.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export async function handleNewOutputEvent(accountId: string, event: NewOutputEv
         syncBalance(account.id)
         addActivityToAccountActivitiesInAllAccountActivities(
             account.id,
-            await new Activity().setFromOutputData(event.output, account)
+            await new Activity().setFromOutputData(event.output, account, event.transactionInputs)
```
Comment (nicole-obrien) [inline packages/shared/lib/core/profile-manager/api/events/newOutput.ts:26]: We also optionally get the transaction so shouldn't we just call Activity().setFromTransaction if its available?

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
We can build optimally from the transaction if it is available.

File: packages/shared/lib/core/profile-manager/api/events/newOutput.ts (hunk 1 above)
Comment (MarkNerdi) [reply packages/shared/lib/core/profile-manager/api/events/newOutput.ts:26]: the transaction we get isn't like the transactions we use in `setFromTransaction`. These `ITransactionPayload`s contain a lot less information than the `Transaction`s, so for now, the only additional information we can use are from the inputs

**Review by nicole-obrien — APPROVED** [review]
Works for me, tested it with alpha group.

**Pull request merged**